### DISCUSSION TITLE
RIST protocol support

### DISCUSSION THREAD

**Discussion by pkviet** [OvenMediaLabs/OvenMediaEngine#1089] [Ideas]
```
RIST is designed to reliably transport video over the Internet for any use case, 
but is typically used for professional media workflows such as news and sports contribution, 
remote production, affiliate distribution and primary distribution.
Source: https://www.rist.tv/
```

It's mainly intended for low-latency transmission over unreliable connections (e.g., commercial internet), so has a similar audience as SRT but probably made different tradeoffs as it was designed collaboratively by multiple companies as opposed to a single vendor (see: https://code.videolan.org/rist/librist#why-do-you-not-improve-srt-rather-than-starting-a-new-project)

Lastly, OBS now supports RIST muxing as of:
obsproject/obs-studio@258365d

and so this transport will soon be available to a much wider range of users client-side, but there is a lack of free software servers supporting RIST ingest.

(Disclaimer: member of the obs-studio dev team here; I coded the support for srt and rist in obs-studio.
This protocol seems a good alternative to srt and in my tests it's very resilient to packet losses.
I unashamedly copied this feature request from here: https://github.com/aler9/rtsp-simple-server/issues/677
The plan as of now is to release RIST support in next release of obs 27.2 by the end of this year.)

**Comment by getroot** [comment]
I'll review RIST.
When applying RIST in OBS, does the internal stream transmit in MPEG2-TS format like SRT? I couldn't find this in the RIST documentation.

**Comment by pkviet** [comment]
Same as srt. Both are agnostic about container, can be anything.
But in practice it's mpegts since broadcast industry is used to mpegts.
In librist repo, you'll find the specs for simple and main profile. The
demo tools ristsender and ristreceiver assume mpegts. Be careful that
ffmpeg implementation of rist is buggy for reception atm, so don't base
your work on it. The librist devs are investigating the issue.
The main difference with srt is that it's based on standards freely
available. (Srt is trying now to do the same.) The capabilities are similar
(arq, fec error correction, bonding, etc)

[a comment by an automated account is not shown]

**Comment by pkviet** [comment]
follow-up from previous post:

- the librist reception bugs on windows have been fixed in librist master 2 or 3 months ago.
- obs-studio can now be used to test against a librist server.
